src/libponyc/codegen/genreference.c

```diff
@@ -312,7 +282,7 @@ static LLVMValueRef gen_digestof_box(compile_t* c, ast_t* type,
 
   LLVMValueRef desc = gendesc_fetch(c, value);
 
-  if(boxed_subtype == SUBTYPE_KIND_BOTH)
+  if((boxed_subtype & SUBTYPE_KIND_UNBOXED) != 0)
```

Comment:
I changed this to `SUBTYPE_KIND_UNBOXED` because the new global `SUBTYPE_KIND_*` enumeration discriminates between numeric and tuple subtypes instead of having them lumped together without any possible distinction like in the previous version of `SUBTYPE_KIND_*` for `digestof`.

This branch should be taken when the subtype set contains an unboxed type and a boxed type (either a numeric type or a tuple). Since we wouldn't be in the function if the set didn't contain a boxed type, we only need to test for the unboxed type.

Using `SUBTYPE_KIND_ALL` would mean that the branch would be taken when the subtype set contains an unboxed type, a numeric type, and a tuple.